src/options.d.ts

```diff
@@ -0,0 +1,16 @@
+import * as Joi from "@hapi/joi";
+
+export const optionsSchema: Joi.SchemaLike
+export const preSpawnSchema: Joi.SchemaLike
+export const spawnSchema: Joi.SchemaLike
+export function parseOptions(options: PartialOptions, filter: string, json: any): any
+export const optionDefaults: IOptions;
+interface IOptions {
+    color: boolean,
+    locations: string[],
+    output: string,
+    raw: boolean,
+    slurp: boolean,
+    sort: boolean,
+}
+export type PartialOptions = Partial<IOptions>
```

Comment:
I'm not familiar with `Partial` in this case, this means that options can be optional and all of their keys are optional as well? or you need to explicitly specify them?

Reply:
Correct. Typescript interface/type members are considered required unless directly specified as optional and they provide a helper to create a member-optional type.